### PR TITLE
PAYARA-1782 add-library command now dynamically adds the files to the…

### PR DESCRIPTION
**Pull request by Cousjava**
… classloader

Note that problems will occur if there are multiple classes with the
same full name, as only the first one to be loaded will be used.

### CODE REVIEW

**Comment by Cousjava** [comment]
jenkins test please

File: nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/CommonClassLoaderServiceImpl.java [hunk 1]
```diff
@@ -168,7 +169,7 @@ private void createCommonClassLoader() {
                     urls.toArray(new URL[urls.size()]), APIClassLoader);
             commonClassLoader.enableCurrentBeforeParent();
         } else {
-            logger.logp(Level.FINE, "CommonClassLoaderManager",
+            logger.logp(Level.SEVERE, "CommonClassLoaderManager",
```
Comment (smillidge) [inline nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/CommonClassLoaderServiceImpl.java]: What's with the increase in Log Level? Why is this SEVERE can an operator who receives the SEVERE log level figure out what to do to fix it?

File: nucleus/deployment/admin/src/main/java/org/glassfish/deployment/admin/AddLibraryCommand.java [hunk 2]
```diff
@@ -117,17 +135,31 @@ public void execute(AdminCommandContext context) {
...
+            ClassLoader commonLoader = commonClsLdr.getCommonClassLoader();
+            CurrentBeforeParentClassLoader loader = null;
+            if (commonLoader instanceof CurrentBeforeParentClassLoader){
+                loader = (CurrentBeforeParentClassLoader) commonLoader;
+            }
 
             for (File libraryFile : files) {
                 if (libraryFile.exists()) {
-                    DeploymentCommandUtils.renameUploadedFileOrCopyInPlaceFile(
+                    logger.log(Level.SEVERE, "ready to add new library");
```
Comment (smillidge) [inline nucleus/deployment/admin/src/main/java/org/glassfish/deployment/admin/AddLibraryCommand.java]: This should be INFO if this is meant for consumption by an administrator or WARNING at most. The Log message also isn't explanatory in any way as to what is required to fix the condition you are logging.

File: nucleus/deployment/admin/src/main/java/org/glassfish/deployment/admin/AddLibraryCommand.java [hunk 3]
```diff
@@ -117,17 +135,31 @@ public void execute(AdminCommandContext context) {
...
+                    File result = DeploymentCommandUtils.renameUploadedFileOrCopyInPlaceFile(
                         libDir, libraryFile, logger, env);
-                    PropertyChangeEvent pe = new PropertyChangeEvent(libDir, 
-                        "add-library", null, libraryFile);
-                    UnprocessedChangeEvent uce = new UnprocessedChangeEvent(
-                        pe, "add-library");
-                    unprocessed.add(uce);
+                    
+                    if (loader != null ){
```
Comment (smillidge) [inline nucleus/deployment/admin/src/main/java/org/glassfish/deployment/admin/AddLibraryCommand.java]: Don't you have to choose a classloader based on the type i.e. ext, common or app?

**Comment by Cousjava** [comment]
jenkins test please

**Comment by Cousjava** [comment]
jenkins test please

**Pull request merged**